tests/std/tests/P0475R1_P0591R4_uses_allocator_construction/test.cpp

```diff
@@ -0,0 +1,181 @@
+        using allocator_type = allocator<int>;
+
+        constexpr OnlyAllocatorConstructible(const allocator<int>&) {}
+    };
+
+    struct DefaultConstructible {
+        constexpr DefaultConstructible() {}
+    };
+
+    using AllocatorArgConstructArgs      = tuple<allocator_arg_t, const allocator<int>&, int&>;
+    using AllocatorConstructArgs         = tuple<int&, const allocator<int>&>;
+    using ConstAllocatorArgConstructArgs = tuple<allocator_arg_t, const allocator<int>&, const int&>;
+    using ConstAllocatorConstructArgs    = tuple<const int&, const allocator<int>&>;
+    using MovedAllocatorArgConstructArgs = tuple<allocator_arg_t, const allocator<int>&, int&&>;
+    using MovedAllocatorConstructArgs    = tuple<int&&, const allocator<int>&>;
+    using OnlyAllocatorArgConstructArgs  = tuple<allocator_arg_t, const allocator<int>&>;
+    using OnlyAllocatorConstructArgs     = tuple<const allocator<int>&>;
+    using DefaultConstructArgs           = tuple<>;
+
+    { // non-pair overload
+        auto tuple1 = uses_allocator_construction_args<int>(alloc, i);
+        static_assert(is_same_v<decltype(tuple1), tuple<int&>>);
+
+        auto tuple2 = uses_allocator_construction_args<AllocatorArgConstructible>(alloc, i);
+        static_assert(is_same_v<decltype(tuple2), AllocatorArgConstructArgs>);
+
+        auto tuple3 = uses_allocator_construction_args<AllocatorConstructible>(alloc, i);
+        static_assert(is_same_v<decltype(tuple3), AllocatorConstructArgs>);
+    }
+
+    { // pair(piecewise_construct_t, tuple, tuple) overload
+        auto tuple4 = uses_allocator_construction_args<pair<int, OnlyAllocatorArgConstructible>>(
+            alloc, piecewise_construct, forward_as_tuple(i), forward_as_tuple());
+        static_assert(
+            is_same_v<decltype(tuple4), tuple<piecewise_construct_t, tuple<int&>, OnlyAllocatorArgConstructArgs>>);
+
+        auto tuple5 = uses_allocator_construction_args<pair<AllocatorConstructible, DefaultConstructible>>(
+            alloc, piecewise_construct, forward_as_tuple(i), forward_as_tuple());
+        static_assert(
+            is_same_v<decltype(tuple5), tuple<piecewise_construct_t, AllocatorConstructArgs, DefaultConstructArgs>>);
+    }
+
+    { // pair() overload
+        auto tuple6 =
+            uses_allocator_construction_args<pair<DefaultConstructible, OnlyAllocatorArgConstructible>>(alloc);
+        static_assert(is_same_v<decltype(tuple6),
+            tuple<piecewise_construct_t, DefaultConstructArgs, OnlyAllocatorArgConstructArgs>>);
+
+        auto tuple7 = uses_allocator_construction_args<pair<OnlyAllocatorConstructible, DefaultConstructible>>(alloc);
+        static_assert(is_same_v<decltype(tuple7),
+            tuple<piecewise_construct_t, OnlyAllocatorConstructArgs, DefaultConstructArgs>>);
+    }
+
+    { // pair(first, second) overload
+        auto tuple4 = uses_allocator_construction_args<pair<int, AllocatorArgConstructible>>(alloc, i, i);

```

Comment:
Although these tests are grouped into different scopes, the tuples were initially numbered in ascending order (1 through 7). Then they start reusing 4 and 5. It would make more sense to keep ascending (this makes errors slightly more readable/searchable).